HelpSource/Guides/UserFAQ.schelp

```diff
@@ -0,0 +1,418 @@
+
+code::
+(
+SynthDef(\simplepink, { |out=0|
+    Out.ar(out, PinkNoise.ar)
+}).add
+)
+::
+
+and create one code::\simplepink:: synth for each channel. Or you could create one SynthDef for each number of channels you expect to use.
+For example if you might use between 1 and 5 channels:
+
+code::
+(
+(1..5).do{ |n|
+SynthDef("simplepink_%".format(n).asSymbol, { |out=0|
+    Out.ar(out, {PinkNoise.ar}.dup(n))
+}).add
+}
+)
+::
+
+Then you'd need to invoke code::\simplepink_4:: or whatever, as appropriate.
+
+
+anchor::Language (client) Issues::
+
+SECTION:: Language (client) issues
+
+subsection:: Calling gui primitives from a SystemClock routine
+
+When calling gui primitives from a SystemClock routine will cause an error:
+
+code::
+SystemClock.sched(0,{ Window.new.front })
+::
+
+code::
+ERROR: Qt: You can not use this Qt functionality in the current thread. Try scheduling on AppClock instead.
+ERROR: Primitive '_QWindow_AvailableGeometry' failed.
+::
+
+To avoid this issue use the AppClock:
+
+code::
+AppClock.sched(0,{ Window.new.front })
+::
+
+or the defer method:
+
+code::
+SystemClock.sched(0,{ { Window.new.front }.defer })
+::
+
+
+subsection:: Binary operations order
+
+Because of the way SuperCollider evaluates expressions, the usual order of execution of mathematical expressions is not respected.
+In SuperCollider everything is an object, and evaluation happens from left to right, so:
+
+code::
+5 + 3 * 2
+::
+
+will evaluate as (5 + 3 ) * 2.
+
+This happens because the expression becomes:
+
+code::
+5.performBinaryOpOnSimpleNumber('+',3).performBinaryOpOnSimpleNumber('*',2)
+::
+
+Therefore, in algebraic expressions parenthesis must be used when left to right orders is not what is desired:
+
+code::
+5 + (3 * 2)
+::
+
+
+anchor::SynthDef Issues::
+
+SECTION:: SynthDef Issues
+
+subsection:: "If" statements inside a SynthDef
+
+It's only a matter of time before a user tries to write something like this in a code::SynthDef::
+
+code::
+SynthDef(\kablooie, { |x = 0|
+    var signal;
+    if(x > 0) {
+        signal = SinOsc.ar
+    } {
+        signal = Saw.ar
+    };
+});
+::
+
+... with the disturbing result: code::ERROR: Non Boolean in test.::
+
+"Non Boolean in test"? But strong::x \> 0:: is a comparison, and surely should produce a Boolean, right?
+
+This should be the first clue that Boolean logic in the server is a very different animal from the so-called "normal" use of conditionals on the client side (in the language).
+
+subsection:: What is a Boolean in the server?
+
+In fact, there is no such thing. The server handles floating-point numbers. It doesn't have strong::true:: or strong::false:: entities.
+
+Since everything in the server is a number, the result of the comparison must also be a number. The server follows the same convention as other DSP environments (Max/MSP, pd etc.):
+
+- code::True:: is represented by 1.0
+- code::False:: is represented by 0.0
+
+subsection:: Why is x \> 0 "non-Boolean" in the "test"?
```

Comment:
I think `>` have been unnecessarily escaped here and in other places in this document. Could you remove these backslashes? Search for `\>`